### PR TITLE
* Add order-now endpoint to store-api

### PR DESCRIPTION
**Pull request by JPruefer**

### CODE REVIEW

File: store-api-server/src/payment/controller/payment.controller.ts [hunk 1]
```diff
@@ -124,60 +125,110 @@ export class PaymentController {
...
+          });
+        return {
+          [firstProvider.provider]: firstProviderIntent,
+          [secondProvider.provider]: secondProviderIntent,
+        };
+      })
+      .catch((err) =>
+        Logger.error(
+          `Unable to create payment intents for payment providers, err: ${err}`,
+        ),
```
Comment (lukeisontheroad) [inline store-api-server/src/payment/controller/payment.controller.ts:230]: I think here we are missing to throw  a `throw new HttpException(...`

Comment (JPruefer) [reply store-api-server/src/payment/controller/payment.controller.ts]: Thanks for the hint, added it!

File: store-api-server/src/payment/service/payment.service.ts [hunk 2]
```diff
@@ -1590,4 +1656,19 @@ WHERE country_short = $1
...
+  getCurrency(provider: PaymentProvider, currency: string): string {
+    switch (provider) {
+      case PaymentProvider.TEZPAY:
+        return 'XTZ';
+      case PaymentProvider.WERT:
+        return 'USD';
+      case PaymentProvider.SIMPLEX:
+        return 'USD';
+      case PaymentProvider.STRIPE:
+        return currency === PaymentProvider.TEZPAY ? 'USD' : currency;
```
Comment (RKlompUU) [inline store-api-server/src/payment/service/payment.service.ts:1669]: This should be `currency === 'XTZ' ? (BASE_CURRENCY === 'XTZ' ? 'USD' : BASE_CURRENCY) : currency`

File: store-api-server/src/payment/service/payment.service.ts [hunk 3]
```diff
@@ -887,11 +953,11 @@ WHERE nft_order_id = $1
     };
   }
 
-  async #registerPayment(
-    dbTx: DbTransaction,
-    nftOrderId: number,
-    paymentIntent: PaymentIntentInternal,
-  ) {
+  async #registerPayment({
```
Comment (RKlompUU) [inline store-api-server/src/payment/service/payment.service.ts:956]: Thanks for fixing these functions that have lots of args to take it as an object instead :+1:

File: store-api-server/src/payment/service/payment.service.ts [hunk 4]
```diff
@@ -1590,4 +1656,19 @@ WHERE country_short = $1
...
+      case PaymentProvider.TEZPAY:
+        return 'XTZ';
+      case PaymentProvider.WERT:
+        return 'USD';
+      case PaymentProvider.SIMPLEX:
+        return 'USD';
+      case PaymentProvider.STRIPE:
+        return currency === PaymentProvider.TEZPAY ? 'USD' : currency;
+      default:
+        return 'USD';
```
Comment (RKlompUU) [inline store-api-server/src/payment/service/payment.service.ts:1671]: This should be `BASE_CURRENCY`

File: store-api-server/src/payment/controller/payment.controller.ts [hunk 5]
```diff
@@ -124,60 +125,114 @@ export class PaymentController {
...
+          });
+        const secondProviderIntent =
+          await this.paymentService.handleCreatePaymentIntent({
+            ...args,
+            paymentProvider: secondProvider.provider,
+            currency: secondProvider.currency,
+          });
+        return {
+          [firstProvider.provider]: firstProviderIntent,
+          [secondProvider.provider]: secondProviderIntent,
```
Comment (RKlompUU) [inline store-api-server/src/payment/controller/payment.controller.ts:224]: This hardcodes this `order-now` to only work with exactly 2 providers. It should work for any number of providers requested. eg it should work when paymentProviders is just `stripe`, or it is `stripe, wert, tezpay`, etc.

File: store-api-server/src/payment/controller/payment.controller.ts [hunk 6]
```diff
@@ -124,60 +125,114 @@ export class PaymentController {
...
+   *    },
+   *    "stripe": {
+   *        ...
+   *    }
+   *  }
+   *  @apiExample {http} Example http request url (make sure to replace $base_url with the store-api-server endpoint):
+   *  $base_url/payment/order-now/10
+   * @apiName orderNow
+   */
+  @Post('/order-now/:id')
```
Comment (RKlompUU) [inline store-api-server/src/payment/controller/payment.controller.ts:183]: I think this `:id` should be `:nftId`

File: store-api-server/src/payment/service/payment.service.ts (hunk 2 above)
Comment (JPruefer) [reply store-api-server/src/payment/service/payment.service.ts]: Should be solved now

File: store-api-server/src/payment/service/payment.service.ts (hunk 4 above)
Comment (JPruefer) [reply store-api-server/src/payment/service/payment.service.ts]: Yes, should be solved now

File: store-api-server/src/payment/controller/payment.controller.ts (hunk 5 above)
Comment (JPruefer) [reply store-api-server/src/payment/controller/payment.controller.ts]: I hope it works now. I could not try it out.

File: store-api-server/src/payment/controller/payment.controller.ts (hunk 6 above)
Comment (JPruefer) [reply store-api-server/src/payment/controller/payment.controller.ts]: Yes, should be solved now

File: store-api-server/src/payment/controller/payment.controller.ts [hunk 7]
```diff
@@ -191,48 +192,71 @@ export class PaymentController {
...
+        request,
+        recreateNftOrder,
+      };
+
+      const firstProvider = providers.shift();
+      if (!firstProvider) {
         throw new HttpException(
-          'failed to create payment intents for payment providers',
+          'No providers provided',
           HttpStatus.INTERNAL_SERVER_ERROR,
```
Comment (RKlompUU) [inline store-api-server/src/payment/controller/payment.controller.ts:216]: this should be a BAD_REQUEST

File: store-api-server/src/payment/controller/payment.controller.ts [hunk 8]
```diff
@@ -191,48 +192,71 @@ export class PaymentController {
...
+            });
+
+          return {
+            [currentProvider.provider]: providerIntent,
+          };
+        }),
+      );
+
+      const allProviderIntents = [
+        ...[{ [firstProvider.provider]: firstIntent }],
```
Comment (RKlompUU) [inline store-api-server/src/payment/controller/payment.controller.ts:241]: cant this line just be `{ [firstProvider.provider]: firstIntent },`?

Comment (JPruefer) [reply store-api-server/src/payment/controller/payment.controller.ts:241]: Array.reduce() only works on arrays. So it needs to be an array.

File: store-api-server/src/payment/controller/payment.controller.ts (hunk 7 above)
Comment (JPruefer) [reply store-api-server/src/payment/controller/payment.controller.ts]: Ups, yes, resolved...

File: store-api-server/src/payment/controller/payment.controller.ts (hunk 8 above)
Comment (JPruefer) [reply store-api-server/src/payment/controller/payment.controller.ts]: You are correct, changed it.

**Pull request merged**